File: ex6 (Yasmin)/ex6-sequencia.hs
sequencia :: (Integer, Integer) -> [Integer]
sequencia (1, m) = [m] -- quando n = 1, adiciona m no início da lista
sequencia (n, m) = sequencia (n-1, m) ++ [m+n-1] -- enquanto n > 1, adiciona M+(N-1) ao início da lista e decrementa n

main = do
    print (sequencia (5, 7))

File: ex6 (Yasmin)/ex6-sequencia.erl
-module(ex6-sequencia).
-export([start/0]).

% quando N = 1, adiciona M no início da lista
sequencia(1, M) -> [M];
sequencia(N, M) when N > 1 ->
     % enquanto N > 1, adiciona M+(N-1) ao início da lista e decrementa N
     sequencia(N-1, M) ++ [M+N-1].

start() ->
    io:fwrite("~p", [sequencia(5, 7)]).
